### PR TITLE
[5.4] add unit tests for form rules

### PR DESCRIPTION
**Pull request by heelc29**
### Summary of Changes

add unit test fore some form rules
also covers updated OptionsRule (#46608)

### Testing Instructions

phpunit

### Expected result AFTER applying this Pull Request

more tests

### CODE REVIEW

File: tests/Unit/Libraries/Cms/Form/Rule/EmailRuleTest.php [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    public function dataTest(): array
+    {
+        $xml = new \SimpleXMLElement('<field
+			name="unittest"
+			type="text"
+			validate="email"
+		/>');
+
+        return [
+            [true, $xml, 'user@example.com'],
```
Comment (brianteeman) [inline tests/Unit/Libraries/Cms/Form/Rule/EmailRuleTest.php:39]: pretty sure that we allow the internal email format user@domain

File: tests/Unit/Libraries/Cms/Form/Rule/BooleanRuleTest.php [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            [true, $xml, 'true'],
+            [true, $xml, 'false'],
+            [true, $xml, '0'],
+            [true, $xml, '1'],
+            [true, $xml, 'TRUE'],
+            [true, $xml, 'FALSE'],
+            [true, $xml, 'True'],
+            [true, $xml, 'False'],
+            [false, $xml, ''],
+            [false, $xml, 'tu re'],
```
Comment (brianteeman) [inline tests/Unit/Libraries/Cms/Form/Rule/BooleanRuleTest.php:48]: you might need to add this intentional typo to [default.extend-words] in ` .github\workflows\typos.toml`

File: tests/Unit/Libraries/Cms/Form/Rule/EmailRuleTest.php (hunk 1 above)
Comment (heelc29) [reply tests/Unit/Libraries/Cms/Form/Rule/EmailRuleTest.php:39]: added

File: tests/Unit/Libraries/Cms/Form/Rule/BooleanRuleTest.php (hunk 2 above)
Comment (heelc29) [reply tests/Unit/Libraries/Cms/Form/Rule/BooleanRuleTest.php:48]: A single space should suffice. I don't want to add an additional exception for this.

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on 8ae628e92bf69b1c20bf875f3abdaffc9bb1dab4<br /><br />Tested by review:
- Reviewed that test cases and test data make sense.
- Checked in logs of unit tests CI action that more tests and assertions have run for this PR compared to the current 5.4-dev branch, and that no new PHP deprecation messages were created by the new test cases.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/46723">issues.joomla.org/tracker/joomla-cms/46723</a>.</sub>

**Comment by 465645** [comment]
I have tested this item :white_check_mark: successfully on 6a1e2c05b29b72033c2282cb2f3fd7b9d9aa134f<br /><br />Tests look correct based on the regex. Quick question: Should we add a test case for whitespace (e.g., `' true '`)? I assume the [BooleanRule](cci:2://file:///c:/xampp/htdocs/joomlapbf/libraries/src/Form/Rule/BooleanRule.php:22:0-39:1) is intended to be strict and reject untrimmed values, so adding that as a negative test case explicitly would confirm that behavior.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/46723">issues.joomla.org/tracker/joomla-cms/46723</a>.</sub>
